### PR TITLE
Replace Account and Path with WalletTools

### PR DESCRIPTION
**Pull request by passabilities**
### Account & Path => Keymanager API
With the update to the keymanager, the use of the Account and Path objects are no longer needed. 
Instead, I created wallet tools that sits on top of they keymanager that the code uses to derive public and private keys and addresses.

### Plugin Util Functions
In an attempt to keep the code modularized and functional, I created some plugin util functions that can be called anywhere in the codebase.

### CODE REVIEW

**Review by passabilities — COMMENTED** [review]
rename folder to utxo engine

File: src/common/utxobased/plugin/utils.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  args.keys[getXprivKey(args)]
+
+export const getXpub = (args: { keys: any, coin: string }): string =>
+  args.keys[getXpubKey(args)]
+
+export const getWalletCoinType = (args: { keys: any }): number =>
+  args.keys.coinType
+
+export const getWalletFormat = (args: { keys: any }): CurrencyFormat =>
+  args.keys.format
```
Comment (passabilities) [inline src/common/utxobased/plugin/utils.ts:82]: return default if no exists

File: src/common/utxobased/plugin/utils.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  `${coin}Xpub`
+
+export const getXpriv = (args: { keys: any, coin: string }): CurrencyFormatKeys =>
+  args.keys[getXprivKey(args)]
+
+export const getXpub = (args: { keys: any, coin: string }): string =>
+  args.keys[getXpubKey(args)]
+
+export const getWalletCoinType = (args: { keys: any }): number =>
+  args.keys.coinType
```
Comment (passabilities) [inline src/common/utxobased/plugin/utils.ts:79]: return default if not exists

Comment (passabilities) [reply src/common/utxobased/plugin/utils.ts:79]: `0`

File: src/common/utxobased/plugin/utils.ts (hunk 1 above)
Comment (passabilities) [reply src/common/utxobased/plugin/utils.ts:82]: `bip32`

File: src/common/utxobased/plugin/utils.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    case BIP43PurposeTypeEnum.WrappedSegwit:
+      keys[getCurrencyFormatFromPurposeType(BIP43PurposeTypeEnum.Segwit)] = seedOrMnemonicToXPriv({
+        ...xprivArgs,
+        type: BIP43PurposeTypeEnum.Segwit
+      })
+      keys[getCurrencyFormatFromPurposeType(BIP43PurposeTypeEnum.WrappedSegwit)] = seedOrMnemonicToXPriv({
+        ...xprivArgs,
+        type: BIP43PurposeTypeEnum.WrappedSegwit
+      })
+      break
```
Comment (passabilities) [inline src/common/utxobased/plugin/utils.ts:129]: get code from local changes

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**